holoviews/plotting/mpl/plot.py

```diff
@@ -963,11 +964,11 @@ def _create_subplots(self, layout, positions, layout_dimensions, ranges, axes={}
                 continue
 
             # Determine projection type for plot
-            components = view.traverse(lambda x: x)
-            projs = ['3d' if isinstance(c, Element3D) else
-                     self.lookup_options(c, 'plot').options.get('projection', None)
-                     for c in components]
-            projs = [p for p in projs if p is not None]
+            projs = self._deep_options(view, 'plot', ['projection'],
+                                      [CompositeOverlay])['projection']
+            if projs:
```

Comment:
Little confused...is this supposed to be ``if not projs``? Seems you would try to look for the options on the elements *if* they couldn't be found at the overlay level, especially since we only allow one projection anyway...